lib/structured_logger/entry.rb

```diff
@@ -0,0 +1,67 @@
+require "structured_logger/event"
+require "structured_logger/metric"
+
+class StructuredLogger
+  class Entry
+
+    def initialize(logger, event=nil, metrics=[], message=nil)
+      @logger = logger
+      @event = event
+      @metrics = metrics
+      @message = message
+    end
+
+    def event(name, options={})
+      Entry.new(@logger, Event.new(name, options), @metrics)
+    end
+
+    def with(options={})
+      Entry.new(@logger, @event.with(options), @metrics)
+    end
+
+    def metric(name, value, type: "counter")
+      metric = Metric.new(name, value, type)
+      Entry.new(@logger, @event, @metrics + [metric])
+    end
+
+    def monitored(subject, body)
+      event = @event.with(monitored: true, subject: subject, body: body)
```

Comment:
Maybe follow in Datadog's footsteps and call it `title` and `text` so that it's not tied to the email paradigm? I would consider email as the implementation detail and "title" and "text" the high-level concepts that can be presented muliple ways.